mvcc/key_index_test.go

```diff
@@ -483,6 +510,22 @@ func TestKeyIndexCompact(t *testing.T) {
 	}
 }
 
+func cloneKeyIndex(ki *keyIndex) *keyIndex {
+	generations := make([]generation, len(ki.generations))
+	for i, gen := range ki.generations {
+		generations[i] = *cloneGeneration(&gen)
+	}
+	return &keyIndex{ki.key, ki.modified, generations}
+}
+
+func cloneGeneration(g *generation) *generation {
+	var tmp []revision
```

Comment:
If `g.revs` is nil, the above code will create a non nil tmp which causes comparison failures.
```sh
--- FAIL: TestKeyIndexCompactAndKeep (0.00s)
	key_index_test.go:451: #0: ki = g: created[{2 0}] ver[3], revs []mvcc.revision{mvcc.revision{main:2, sub:0}, mvcc.revision{main:4, sub:0}, mvcc.revision{main:6, sub:0}}
		g: created[{8 0}] ver[3], revs []mvcc.revision{mvcc.revision{main:8, sub:0}, mvcc.revision{main:10, sub:0}, mvcc.revision{main:12, sub:0}}
		g: created[{14 0}] ver[3], revs []mvcc.revision{mvcc.revision{main:14, sub:0}, mvcc.revision{main:14, sub:1}, mvcc.revision{main:16, sub:0}}
		g: created[{0 0}] ver[0], revs []mvcc.revision(nil)
		, want g: created[{2 0}] ver[3], revs []mvcc.revision{mvcc.revision{main:2, sub:0}, mvcc.revision{main:4, sub:0}, mvcc.revision{main:6, sub:0}}
		g: created[{8 0}] ver[3], revs []mvcc.revision{mvcc.revision{main:8, sub:0}, mvcc.revision{main:10, sub:0}, mvcc.revision{main:12, sub:0}}
		g: created[{14 0}] ver[3], revs []mvcc.revision{mvcc.revision{main:14, sub:0}, mvcc.revision{main:14, sub:1}, mvcc.revision{main:16, sub:0}}
		g: created[{0 0}] ver[0], revs []mvcc.revision{}
```